File: JS/Groups of Special-Equivalent Strings.js
/**
 * @param {string[]} A
 * @return {number}
 */
var numSpecialEquivGroups = function(A) {
    let set = new Set();
    for( let words of A ){
        let temp =  sort(words);
        set.add(temp);
    }
    return set.size
};    
let sort = function (s){
    let odd='',even='';
    for(let i=0;i<s.length; ++i){
        if(i % 2){
           even += s[i];
         }else{
             odd += s[i];
         }
    }
    even=even.split('').sort().join('');
    odd=odd.split('').sort().join('');
    return even+odd;
}    

File: C++/Groups of Special-Equivalent Strings.cpp
class Solution {
public:
    int numSpecialEquivGroups(vector<string>& A) {
        unordered_set<string> s;
        for(auto str : A){
            string odd,even;
            for(int i=0;i<str.size();++i){
                if(i % 2){
                    even += str[i];
                }else{
                    odd += str[i];
                }
            }
           sort(even.begin(),even.end());
           sort(odd.begin(),odd.end());
           s.insert(even+odd);
        }
        return s.size();
    }
};